### PR TITLE
[OWP-30] Update deprecated APIs in iOS

### PR DESCRIPTION
**Pull request by swetha-gadiraju**
- Updated deprecated APIs in iOS SDK
- Updated min iOS version to 15 in example app
- Fixed authenticateDevice API issue in example app for iOS

### CODE REVIEW

File: ios/Classes/NativeBridge/OneginiModuleSwift.swift [hunk 1]
```diff
@@ -18,26 +18,22 @@ public class OneginiModuleSwift: NSObject {
...
-        ONGClientBuilder().setAdditionalResourceUrls(additionalResourceUrls ?? [])
-        ONGClientBuilder().build()
-        ONGClient.sharedInstance().start { result, error in
-            if let error = error {
-                let mappedError = ErrorMapper().mapError(error)
-                callback(.failure(mappedError.flutterError()))
-                return
+        _ = ClientBuilder().setHttpRequestTimeout(TimeInterval(Double(httpConnectionTimeout ?? 5)))
+        _ = ClientBuilder().setAdditionalResourceUrls(additionalResourceUrls ?? [])
+        ClientBuilder().buildAndWaitForProtectedData { client in
```
Comment (kfa-onewelcome) [inline ios/Classes/NativeBridge/OneginiModuleSwift.swift:23]:
```
let builder = ClientBuilder()

builder.
builder.
```

File: ios/Classes/NativeBridge/OneginiModuleSwift.swift [hunk 2]
```diff
@@ -18,26 +18,22 @@ public class OneginiModuleSwift: NSObject {
...
+            client.start { error in
+                if let error = error {
```
Comment (kfa-onewelcome) [inline ios/Classes/NativeBridge/OneginiModuleSwift.swift:25]: `if let error { }`

File: example/lib/screens/user_screen.dart [hunk 3]
```diff
@@ -435,8 +435,7 @@ class Info extends StatefulWidget {
 
 class _InfoState extends State<Info> {
   Future<ApplicationDetails> _getApplicationDetails() async {
-    await Onegini.instance.userClient
-        .authenticateDevice(["read", "write", "application-details"]);
+    await Onegini.instance.userClient.authenticateDevice(["application-details"]);
```
Comment (kfa-onewelcome) [inline example/lib/screens/user_screen.dart:438]: if `read` and `write` is needed, you can workaround the issue with
 `.authenticateDevice(["read write application-details"])`

**(but pls double check if that workaround works)**

The fix for that is being addressed

**Review by kfa-onewelcome — CHANGES_REQUESTED** [review]

File: ios/Classes/NativeBridge/OneginiModuleSwift.swift (hunk 1 above)
Comment (swetha-gadiraju) [reply ios/Classes/NativeBridge/OneginiModuleSwift.swift]: Updated

File: ios/Classes/NativeBridge/OneginiModuleSwift.swift (hunk 2 above)
Comment (swetha-gadiraju) [reply ios/Classes/NativeBridge/OneginiModuleSwift.swift]: Updated

File: example/lib/screens/user_screen.dart (hunk 3 above)
Comment (swetha-gadiraju) [reply example/lib/screens/user_screen.dart:438]: This workaround works but read/write not required here in example app to show application-details. So didn't add back.

**Review by kfa-onewelcome — APPROVED** [review]

**Pull request merged**